insights/parsers/numastat.py

```diff
@@ -0,0 +1,335 @@
+                 'calculated_hugepages_surp': 0.0,
+                 'calculated_hugepages_total': 56320.0,
+                 'dirty': 0.0,
+                 'filepages': 8440.29,
+                 'hugepages_free': 51,
+                 'hugepages_surp': 0,
+                 'hugepages_total': 55,
+                 'inactive': 3422.76,
+                 'inactive(anon)': 1.23,
+                 'inactive(file)': 3421.53,
+                 'kernelstack': 9.97,
+                 'mapped': 117.59,
+                 'memfree': 28190.93,
+                 'memtotal': 95257.24,
+                 'memused': 67066.32,
+                 'mlocked': 96.62,
+                 'nfs_unstable': 0.0,
+                 'pagetables': 10.84,
+                 'shmem': 3.41,
+                 'slab': 716.69,
+                 'sreclaimable': 446.41,
+                 'sunreclaim': 270.28,
+                 'unevictable': 96.62,
+                 'writeback': 0.0,
+                 'writebacktmp': 0.0
+              },
+              'node1': {
+                 'active': 2210.93,
+                 'active(anon)': 369.71,
+                 'active(file)': 1841.22,
+                 'anonhugepages': 0,
+                 'anonpages': 393.97,
+                 'bounce': 0.0,
+                 'calculated_anonhugepages': 10.0,
+                 'calculated_hugepages_free': 52224.0,
+                 'calculated_hugepages_surp': 0.0,
+                 'calculated_hugepages_total': 56320.0,
+                 'dirty': 0.14,
+                 'filepages': 4427.57,
+                 'hugepages_free': 51,
+                 'hugepages_surp': 0,
+                 'hugepages_total': 55,
+                 'inactive': 2570.12,
+                 'inactive(anon)': 0.26,
+                 'inactive(file)': 2569.87,
+                 'kernelstack': 10.03,
+                 'mapped': 103.52,
+                 'memfree': 31256.51,
+                 'memtotal': 96732.4,
+                 'memused': 65475.89,
+                 'mlocked': 40.25,
+                 'nfs_unstable': 0.0,
+                 'pagetables': 7.59,
+                 'shmem': 2.55,
+                 'slab': 541.11,
+                 'sreclaimable': 321.7,
+                 'sunreclaim': 219.42,
+                 'unevictable': 40.25,
+                 'writeback': 0.0,
+                 'writebacktmp': 0.0
+              }
+            }
+
+       ``meminfo`` (dict): Dicitonnary of nodes with MemInfo objects:
+
+            {
+             'Total': <insights.parsers.numastat.NUMAMemInfo object at 0x7f42d8646278>,
+             'node0': <insights.parsers.numastat.NUMAMemInfo object at 0x7f42d8633860>,
+             'node1': <insights.parsers.numastat.NUMAMemInfo object at 0x7f42d8633f28>},
+            }
+
+    """
+
+    # Not sure how to get this info from here
+    # so I made it a constant. This is pretty
+    # much standard anyways, at least in OSP
+    HUPAGESIZE = 1048576
+
```

Comment:
This looks OK.  Could you explain the comment?

Reply:
When we look at `/proc/meminfo`, the HP are shown as numbers, but we can determine the amount of memory allocated by multiplying by the `Hugepagesize` attribute
```
[root@ess13latest-scpu-0 ~]# grep -P "Huge[pP]ages(ize:|_(Total|Free))" /proc/meminfo
HugePages_Total:     110
HugePages_Free:      102
Hugepagesize:    1048576 kB
```
When we look at `numastat -m`, we only have the amount of memory allocated to HP. We don't know how many HP or their size. 
```
[root@ess13latest-scpu-0 ~]# numastat -m | grep -P "Huge[pP]ages(ize:|_(Total|Free))"
HugePages_Total         56320.00        56320.00       112640.00
HugePages_Free          52224.00        52224.00       104448.00
```

The `MemInfo` parser is expecting the number of HP and not the amount of memory. So I need to guess the number of HP based on missing variables. Ideally, I'd grab the HP size from `/proc/meminfo`, from inside that parser but I'm not sure how to do this.

I think 95% of the deployments are using HP size of 1048576 but I've seen deployments with smaller "huge" page size like 4096.